wasm/read.go

```diff
@@ -5,20 +5,34 @@
 package wasm
 
 import (
+	"bytes"
 	"encoding/binary"
 	"io"
 
 	"github.com/go-interpreter/wagon/wasm/leb128"
 )
 
-func readBytes(r io.Reader, n int) ([]byte, error) {
-	bytes := make([]byte, n)
-	_, err := io.ReadFull(r, bytes)
-	if err != nil {
-		return bytes, err
+// to avoid memory attack
+const maxInitialCap = 10 * 1024
+
+func getInitialCap(count uint32) uint32 {
+	if count > maxInitialCap {
+		return maxInitialCap
 	}
+	return count
+}
 
-	return bytes, nil
+func readBytes(r io.Reader, n int) ([]byte, error) {
+	if n == 0 {
+		return nil, nil
+	}
+	limited := io.LimitReader(r, int64(n))
```

Comment:
I don't understand what's the advantage of using an `io.LimitedReader` + a `bytes.Buffer` over a simple `[]byte` and `io.ReadFull`.
could you explain this change?

Reply:
`n` is controlled by untrusted source. The attacker can actually send you 10 bytes, but set `n` to 1GB. so directly allocate `n` bytes memory is not safe. Use `buf.ReadFrom`, it will read the data by chunks, enlarge the internal buf on demand. In this way, we will avoid this attack, since when we read out 10 bytes, it will hit EOF.

Reply:
yes, this I understood :)
but I was under the assumption we'd always pass `n` through `getInitialCap` (or some other sanity check) before passing it to `readBytes`.

Reply:
`getInitialCap` is just a hint function to avoid some internal reallocation, so I just name it `inital`, not `max`. It does not impose any constraints on `n`.

Reply:
actually, `make([]Value, 0, getInitialCap(count))` can just be replaced with `make([]Value, 0)` if we do not care about that optimization.